### PR TITLE
Feat/cart

### PR DESCRIPTION
**Pull request by vikram1995**

### CODE REVIEW

File: src/components/cart/cart.js [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+import CartList from "./cartList";
+
+function Cart(props) {
+  const {isCartUpdated } = props;
+  const [cartCount, setCartCount] = useState(0);
+  const [cart, setCart] = useState([]);
+
+  useEffect(() => {
+    let cart = localStorage.getItem("cart");
+    if (cart !== "undefined" && cart !== null) {
```
Comment (ergauravgoyal) [inline src/components/cart/cart.js]: we should not compare a value with undefined directly.. Use lodash methods instead

File: src/components/cart/cart.js [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
+      console.log(cart);
+      cart = JSON.parse(cart);
```
Comment (ergauravgoyal) [inline src/components/cart/cart.js]: We can write a method in utils and use it across

File: src/components/cart/cart.js [hunk 3]
```diff
@@ -1,11 +1,38 @@
...
+      setCart(cart);
+      setCartCount(cart.length);
```
Comment (ergauravgoyal) [inline src/components/cart/cart.js]: we can use size of lodash

File: src/components/header/header.js [hunk 4]
```diff
@@ -21,21 +21,32 @@ function header() {
...
+            <Logo>Fashion Town</Logo>
           </Link>
         </Col>
 
         <Col className="gutter-row" span={5}>
           <NavBarGroup>
             <Space size={"large"}>
               <NavBarItems>
-                <NavText>MEN</NavText>
+                <NavLink to={`/shop?idealFor=Men&productCategory=Clothing`}>
```
Comment (ergauravgoyal) [inline src/components/header/header.js]: we should store routes in config file

File: src/components/main.js [hunk 5]
```diff
@@ -15,8 +17,18 @@ export class Main extends Component {
           <Route path="/" element={<LandingPage />} />
           <Route path="signIn" element={<SignIn />} />
           <Route path="SignUp" element={<SignUp />} />
+          <Route path="shop" element={<FilterAndProductListing />} />
+          <Route path="product/:id" element={<ProductDetailsPage />} />
+          <Route
+            path="*"
+            element={
+              <main style={{ padding: "1rem" }}>
+                <p>There's nothing here!</p>
```
Comment (ergauravgoyal) [inline src/components/main.js]: change message to "We couldn't find the page"

File: src/components/productCard/productCard.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      onMouseOver={handleAutoPlay}
+      onMouseOut={handleAutoPlay}
+      loading={isLoading}
+      bodyStyle={{ width: 200, padding: "5px" }}
+      cover={
+        <Carousel ref={carousel} autoplay={autoPlayFlag} autoplaySpeed={1000}>
+          {productData.images.map((image, index) => {
+            return (
+              <div key={index}>
+                {index === 0 ? (
```
Comment (ergauravgoyal) [inline src/components/productCard/productCard.js]: use a common method to check undefined, null, 0 or length as 0

File: src/components/productCard/productCard.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+                  <Image
+                    preview={false}
+                    width={200}
+                    src={image}
+                    placeholder={<Skeleton active={true} />}
+                  />
+                ) : (
+                  <Image preview={false} width={200} src={image} />
```
Comment (ergauravgoyal) [inline src/components/productCard/productCard.js]: we dont need to pass false value, we can omit false value for props

File: src/components/productDetails/productDescription.js [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+      message.error("Please Select Size");
+      return false;
+    }
+    return true;
+  };
+
+  const addToCart = () => {
+    if (isSizeSelected()) {
+      const { dispatch, isCartUpdated } = props;
+      let cart = localStorage.getItem("cart");
```
Comment (ergauravgoyal) [inline src/components/productDetails/productDescription.js]: define a method in utils and use it across

File: src/components/productDetails/productDescription.js [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+      if (!cart) {
+        localStorage.setItem("cart", JSON.stringify([product]));
```
Comment (ergauravgoyal) [inline src/components/productDetails/productDescription.js]: define a method in utils and use it across

File: src/components/productDetails/productDescription.js [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+        message.success("Item added to cart");
+      } else {
+        cart = JSON.parse(cart);
+        const updatedCart = getUpdatedCart(cart, product);
+        localStorage.setItem("cart", JSON.stringify(updatedCart));
```
Comment (ergauravgoyal) [inline src/components/productDetails/productDescription.js]: define a method in utils and use it across

File: src/components/productDetails/productDetailsStyledComponent.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+  font-size: 1em;
+`;
+
+const ProductDescriptionWrapper = styled.div`
+    padding: 40px;
+    padding-right:25%;
+`
+const TaxText = styled.p`
+    color: green;
+    font-size: 0.9em;
```
Comment (ergauravgoyal) [inline src/components/productDetails/productDetailsStyledComponent.js]: we should not change font-size at multiple places. use a common typography across

File: src/components/productDetails/productDetailsStyledComponent.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+`
+
+const HorizontalLine = styled.div`
+    border: 1px solid #e8e8e8;
+    width: 100%;
+    height: 0px;
```
Comment (ergauravgoyal) [inline src/components/productDetails/productDetailsStyledComponent.js]: we dont need height with border

File: src/services/apolloClient.js [hunk 13]
```diff
@@ -13,7 +13,7 @@ const errorLink = onError(({ graphqlErrors, networkErrors }) => {
   }
 });
 
-const link = from([errorLink, new HttpLink({ uri: "http://localhost:4000/" })]);
+const link = from([errorLink, new HttpLink({ uri: "http://localhost:5001/" })]);
```
Comment (ergauravgoyal) [inline src/services/apolloClient.js]: check the port number for error

**Review by ergauravgoyal — CHANGES_REQUESTED** [review]
Hey @vikram1995 : Can you please address the review comments

File: src/components/header/header.js (hunk 4 above)
Comment (vikram1995) [reply src/components/header/header.js]: moved to config

File: src/components/main.js (hunk 5 above)
Comment (vikram1995) [reply src/components/main.js]: Added a page for it

File: src/components/productDetails/productDescription.js (hunk 9 above)
Comment (vikram1995) [reply src/components/productDetails/productDescription.js]: used redux persist

File: src/components/productDetails/productDescription.js (hunk 10 above)
Comment (vikram1995) [reply src/components/productDetails/productDescription.js]: used redux persist

**Pull request closed without merging**